editoast/src/views/infra/mod.rs

```diff
@@ -1083,6 +1110,38 @@ async fn find_track_names(
     )))
 }
 
+async fn populate_op_geo(
+    db_pool: Arc<DbConnectionPoolV2>,
+    infra: &mut Infra,
+    infra_cache: &InfraCache,
+    operational_points: &[Vec<OperationalPoint>],
+) -> Result<Vec<Vec<RelatedOperationalPoint>>> {
+    let op_ids = operational_points
+        .iter()
+        .flatten()
+        .map(|op| op.id.as_str())
+        .collect_vec();
+
+    infra.refresh(db_pool.clone(), false, infra_cache).await?;
+    let geo_points =
+        OperationalPointLayer::get(&mut db_pool.get().await?, infra.id, &op_ids).await?;
+
+    Ok(operational_points
+        .iter()
+        .map(|ops| {
+            ops.iter()
+                .map(|op| RelatedOperationalPoint {
+                    op: op.clone(),
+                    geo: geo_points
+                        .get(op.id.as_str())
+                        .unwrap_or_else(|| panic!("No geo point returned for OP ID {}", op.id))
```

Comment:
This is possible, let's say you have an operational point without parts, it won't have a geometry.
Same if an operational point is in an error state (attached to not found tracks).
We can return an empty `MultiPoint`, or just an `Option<MultiPoint>`

Reply:
Hm, `SearchResultItemOperationalPoint` has a non-optional `GeoJsonPoint`. I wonder what happens for invalid/missing OP parts there: are these just missing from the `infra_layer_operational_point` table? As a result, are these never returned by the search API? Should we do the same here?

Reply:
Alright, opened https://github.com/OpenRailAssociation/osrd/issues/12575